web/third/bootstrap-typeahead/typeahead.js

```diff
@@ -495,6 +502,13 @@ import {get_string_diff} from "../../src/util";
               hideOnEmpty = true;
             }
           }
+          if (this.options.openInputFieldOnKeyUp !== null && !this.shown) {
+            // If the typeahead isn't shown yet, the `lookup` call will open it.
+            // Here we make do a callback to the input field before we open the
+            // lookahead in case it needs to make UI changes first (e.g. widening
+            // the search bar).
+            this.options.openInputFieldOnKeyUp();
+          }
```

Comment:
Hmm, why the change here rather than in `.show()`? That would be more symmetric with the new `.hide()`  hook you're adding.

Reply:
The function is used in #24345 (linked in PR description) -- would it be better to keep it with the PR, or does having it in a separate prep PR help with review? Still learning what works best for you :)

Reply:
> Hmm, why the change here rather than in .show()? That would be more symmetric with the new .hide() hook you're adding. (I'm worried that placing this not next to where .shown gets set risks there being bugs where it isn't called correctly).

That's a great question, and I think I tried it in `show` and decided not to for some reason I've since forgotten, but I can't seem to replicate an issue now so my guess is that reason isn't relevant anymore. I can move it to `show`, which would be nicer as you point out!